### DISCUSSION TITLE
Velero 1.15.0 - DataUpload CR sometimes not updated / finalized or Bytes missing?

### DISCUSSION THREAD

**Discussion by realjump** [vmware-tanzu/velero#8492] [Community support Q&A] [answered]
Hi everyone, we have noticed that after upgrading to Velero 1.15.0, DataUploads sometimes “stop” at a certain number of bytes and then still complete successfully. It can also be seen that the typical events “Data-Path-Completed / Stopped” are missing in these CR resources.

From the logs of the DataUpload pod you can only recognize a “Timeout waiting for assured events processed” error as a difference to the logs of a “correct” DataUpload pod.

I am attaching some excerpts from the same backup run as an example. Can anyone say whether bytes were really not uploaded here or whether the CR resource in this example is simply not being updated / finalized correctly?

Another difference was that an error was ignored due to “unknown or unsupported entry” (see screenshot). However, I could also partially see this with DataUpload CRs, which appear correct, i.e. bytes_done = total_bytes.

### Correctly appearing DataUpload CR:

![correct-cr](https://github.com/user-attachments/assets/2112bab8-967e-4de8-bc0b-b720282e3133)

### Correctly appearing DataUpload - Logs from DataUpload Pod:

![correct-logs](https://github.com/user-attachments/assets/1de3592b-457e-4885-9c9e-71a950b4af53)

### DataUpload CR not appearing correctly:

![incorrect-cr](https://github.com/user-attachments/assets/e6995ec4-e8e0-4226-a5b5-d7b7ba81fc46)

### DataUpload CR not appearing correctly - Logs from DataUpload Pod:

![timeout_backup](https://github.com/user-attachments/assets/484d6a44-231a-4ae3-a46b-e6ee304611e8)

**Addition:** I have checked this for all DataUploads and it always seems to be the case that the corresponding timeout is visible. The DataUpload pods are then still visible as Completed for 1-2 minutes and then the status of the CR also changes from InProgress to Completed, but again with bytes_done < total_bytes.

> time="2024-12-06T09:22:40Z" level=info msg="Data path service completed" du=backupname-backup-20241206063243-zlxtm logSource="pkg/cmd/cli/datamover/backup.go:243"
> 
> time="2024-12-06T09:22:40Z" level=info msg="Waiting sentinel before shutdown" logSource="pkg/util/kube/event.go:138"
> 
> time="2024-12-06T09:23:40Z" level=warning msg="Timeout waiting for assured events processed" logSource="pkg/util/kube/event.go:146"
> 
> time="2024-12-06T09:23:40Z" level=info msg="Closing FileSystemBR" dataupload=backupname-backup-20241206063243-zlxtm logSource="pkg/datapath/file_system.go:145" user=backupname-backup-20241206063243-zlxtm
> 
> time="2024-12-06T09:23:40Z" level=info msg="FileSystemBR is closed" dataupload=backupname-backup-20241206063243-zlxtm logSource="pkg/datapath/file_system.go:151" user=backupname-backup-20241206063243-zlxtm
> 
> time="2024-12-06T09:23:40Z" level=info msg="Data path service is shut down" du=backupname-backup-20241206063243-zlxtm logSource="pkg/cmd/cli/datamover/backup.go:247"

**Comment by Lyndon-Li** [comment, marked as answer]
@realjump Please open an issue for the same. Please also clarify how many DUs are there when the problem happened. And also clarify your cluster nodes' capacity, i.e., what is the CPU usage in the node where the problem happened.

**Reply by realjump** [reply]
Thanks - done https://github.com/vmware-tanzu/velero/issues/8497. 
Let me know if you need any further information.
